third_party/move/move-compiler-v2/src/file_format_generator/peephole_optimizer/inefficient_loads.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module contains a window peephole optimizer for the Move bytecode.
+//! As with all peephole optimizers here, it assumes that the bytecode is valid.
+//!
+//! This optimizer addresses a commonly appearing pattern when involving loads.
+//!
+//! The pattern is:
+//! 1. Load a constant into the stack.
+//! 2. Store the constant into a local `u`.
+//! 3. A (possibly empty) sequence of instructions that do not involve `u`,
+//!    which we name `sequence`. Currently, the only instructions that can
+//!    involve `u` are: `CopyLoc`, `MoveLoc`, `StLoc`, `ImmBorrowLoc`,
+//!    and `MutBorrowLoc`.
+//! 4. A `MoveLoc` of `u`.
+//!
+//! This pattern can be replaced with:
+//! 1. `sequence`.
+//! 2. Load the constant into the stack.
+//!
+//! This transformation leaves the stack in the same state.
+//! The local `u` in the original code has been moved from, so later code
+//! cannot use it without a subsequent store.
+//! So, skipping the store to `u` is safe.
+
+use crate::file_format_generator::peephole_optimizer::optimizers::WindowOptimizer;
+use move_binary_format::file_format::Bytecode;
+
+/// An optimizer for inefficient loads.
+pub struct InefficientLoads;
+
+impl InefficientLoads {
+    // We need at least 3 instructions, corresponding to points 1, 2, and 4 in the pattern
+    // described in the module documentation.
```

Comment:
module documentation --> file documentation, above.

Reply:
It is indeed the module documentation (made using `//!` at the beginning of the file).

I'll add saying "at the top of the file" to clarify, but we have used this terminology elsewhere in the code base in the same sense as here.